test/k8sT/Services.go

```diff
@@ -580,164 +581,159 @@ var _ = Describe("K8sServicesTest", func() {
+				}...)
+			}
+			if testSecondaryNodePortIP {
+				secondaryK8s1IPv4, _ = getIPv4Andv6AddrForIface(k8s1NodeName, helpers.SecondaryIface)
+				secondaryK8s2IPv4, _ = getIPv4Andv6AddrForIface(k8s2NodeName, helpers.SecondaryIface)
 
-				// From pod via remote cilium_host
-				httpURL = getHTTPLink(remoteCiliumHostIPv4, data.Spec.Ports[0].NodePort)
-				tftpURL = getTFTPLink(remoteCiliumHostIPv4, data.Spec.Ports[1].NodePort)
-				testCurlRequest(testDSClient, httpURL)
-				testCurlRequest(testDSClient, tftpURL)
+				testURLsFromHosts = append(testURLsFromHosts, []string{
+					getHTTPLink(secondaryK8s1IPv4, data.Spec.Ports[0].NodePort),
+					getTFTPLink(secondaryK8s1IPv4, data.Spec.Ports[1].NodePort),
 
-				httpURL = getHTTPLink("::ffff:"+remoteCiliumHostIPv4, data.Spec.Ports[0].NodePort)
-				tftpURL = getTFTPLink("::ffff:"+remoteCiliumHostIPv4, data.Spec.Ports[1].NodePort)
-				testCurlRequest(testDSClient, httpURL)
-				testCurlRequest(testDSClient, tftpURL)
+					getHTTPLink(secondaryK8s2IPv4, data.Spec.Ports[0].NodePort),
+					getTFTPLink(secondaryK8s2IPv4, data.Spec.Ports[1].NodePort),
+				}...)
+			}
 
+			testURLsFromOutside := []string{}
+			if testFromOutside {
+				// These are tested from external node which does not run
+				// cilium-agent (so it's not a subject to bpf_sock)
+				testURLsFromOutside = []string{
+					getHTTPLink(k8s1IP, data.Spec.Ports[0].NodePort),
+					getTFTPLink(k8s1IP, data.Spec.Ports[1].NodePort),
+
+					getHTTPLink(k8s2IP, data.Spec.Ports[0].NodePort),
+					getTFTPLink(k8s2IP, data.Spec.Ports[1].NodePort),
+				}
 				if testSecondaryNodePortIP {
-					httpURL = getHTTPLink(secondaryK8s1IPv4, data.Spec.Ports[0].NodePort)
-					tftpURL = getTFTPLink(secondaryK8s1IPv4, data.Spec.Ports[1].NodePort)
-					doRequests(httpURL, count, k8s1NodeName)
-					doRequests(tftpURL, count, k8s1NodeName)
-					if testFromOutside {
-						doRequestsFromThirdHost(httpURL, count, false)
-						doRequestsFromThirdHost(tftpURL, count, false)
-					}
+					testURLsFromOutside = append(testURLsFromOutside, []string{
+						getHTTPLink(secondaryK8s1IPv4, data.Spec.Ports[0].NodePort),
+						getTFTPLink(secondaryK8s1IPv4, data.Spec.Ports[1].NodePort),
 
-					httpURL = getHTTPLink(secondaryK8s2IPv4, data.Spec.Ports[0].NodePort)
-					tftpURL = getTFTPLink(secondaryK8s2IPv4, data.Spec.Ports[1].NodePort)
-					doRequests(httpURL, count, k8s2NodeName)
-					doRequests(tftpURL, count, k8s2NodeName)
-					if testFromOutside {
-						doRequestsFromThirdHost(httpURL, count, false)
-						doRequestsFromThirdHost(tftpURL, count, false)
-					}
+						getHTTPLink(secondaryK8s2IPv4, data.Spec.Ports[0].NodePort),
+						getTFTPLink(secondaryK8s2IPv4, data.Spec.Ports[1].NodePort),
+					}...)
 				}
+			}
+
+			count := 10
+			for _, url := range testURLsFromPods {
+				wg.Add(1)
+				go func(url string) {
+					defer wg.Done()
+					testCurlRequest(testDSClient, url)
```

Comment:
I think you will need to increment `ExpectWithOffset` offset (first argument) in each helper test function called in goroutines so we get better context information about error if assertion fails.